doc/datasets/adult_census.rst

```diff
@@ -0,0 +1,22 @@
+
+.. _adult_census:
+
+Adult census
+============
+
+The samples in this dataset was extracted by Barry Becker from the 1994 Census
```

Comment:
I would rather start this section with what the data is, and what its format is, rather than history about its creation...